md/README.md

```diff
@@ -0,0 +1,90 @@
+# Mention Detection
+
+A mention detection module that annoatates all mentions of the given TextAnnoatation
+
+## Introduction
+
+### Mention Detection
+
+A mention detection module aims to annotate all the mentions of different entities in the given texts. In this package, we took the approach of detection all mention heads first, then predict mention extents for all the predicted mention heads.
+
+### Head Detection
+
+This package tagged all the tokens in a text into either "B/I/O" or "B/I/O/L/U" schema, then train/test on each single token. After evaluating all tokens, we interpret this representation back to mentions.
+
+### Extent Detection
+
+This package does extent classification after detecting mention heads. In the training process, each token in the extent to both right and left of a given mention head is formed into pairs to train a binary classifier, saying whether the token is in the extent or not. To add negative examples, one token to the left and right of the mention extent is added. In the evaluating process, given a mention head, the package forms pairs with tokens to the left and right, until finding a token that is predicted to be not in the extent.
+
+## Results
+
+Head Boundary
+
+| Train\Eval (F1) | ACE  | ERE  |
+|-----------------|------|------|
+| ACE             | 89.6 | 83.8 |
+| ERE             | 81.7 | 86.7 |
+| ACE+ERE         | 88.4 | 86.5 |
+
+Extent boundary accuracy given Head
+
+ACE: 89.45
+
+## Using Annotator
+
+```java
+TextAnnoatation to_be_annoatated;
+MentionAnnotator mentionAnnotator = new MentionAnnotator();
+mentionAnnotator.addView(to_be_annotated);
+List predictedMentions = to_be_annotated.getView("MENTION").getConstituents();
```

Comment:
Can you change it so that it contains imports too? 
Also, how you get from text to TextAnnotation. See this example from NER: 
```
import edu.illinois.cs.cogcomp.core.datastructures.textannotation.TextAnnotation;
import edu.illinois.cs.cogcomp.core.utilities.configuration.ResourceManager;
import edu.illinois.cs.cogcomp.nlp.utility.TokenizerTextAnnotationBuilder;
import edu.illinois.cs.cogcomp.annotation.TextAnnotationBuilder;
import edu.illinois.cs.cogcomp.core.datastructures.ViewNames;
import edu.illinois.cs.cogcomp.ner.NERAnnotator;
import edu.illinois.cs.cogcomp.nlp.tokenizer.StatefulTokenizer;
import edu.illinois.cs.cogcomp.ner.LbjTagger.*;
import java.io.IOException;

import java.util.Properties;

// Filename: App.java
public class App
{
    public static void main( String[] args ) throws IOException
    {
        String text1 = "Good afternoon, gentlemen. I am a HAL-9000 "
            + "computer. I was born in Urbana, Il. in 1992";

        String corpus = "2001_ODYSSEY";
        String textId = "001";

        // Create a TextAnnotation using the LBJ sentence splitter
        // and tokenizers.
        TextAnnotationBuilder tab;
        // don't split on hyphens, as NER models are trained this way
        boolean splitOnHyphens = false;
        tab = new TokenizerTextAnnotationBuilder(new StatefulTokenizer(splitOnHyphens));

        TextAnnotation ta = tab.createTextAnnotation(corpus, textId, text1);

        NERAnnotator co = new NERAnnotator(ViewNames.NER_CONLL);
        co.getView(ta);

        System.out.println(ta.getView(ViewNames.NER_CONLL));
    }
}
```